src/replica/replica_disk_migrator.cpp

```diff
@@ -148,12 +152,141 @@ bool replica_disk_migrator::check_migration_args(const replica_disk_migrate_requ
+        dwarn_replica("disk migration(origin={}, target={}) target replica dir({}) has existed, it "
+                      "will be deleted",
+                      req.origin_disk,
+                      req.target_disk,
+                      _target_replica_dir);
+        utils::filesystem::remove_path(_target_replica_dir);
+    }
+
+    //  _target_replica_data_dir = /root/gpid.app_type.disk.balance.tmp/data/rdb, it will update to
+    //  /root/target/gpid.app_type/data/rdb in replica_disk_migrator::update_replica_dir finally
+    _target_data_dir = utils::filesystem::path_combine(_target_replica_dir, kDataDirFolder);
+    if (!utils::filesystem::create_directory(_target_data_dir)) {
+        derror_replica(
+            "disk migration(origin={}, target={}) create target temp data dir({}) failed",
+            req.origin_disk,
+            req.target_disk,
+            _target_data_dir);
+        reset_status();
+        return false;
+    }
+
+    return true;
+}
+
+// THREAD_POOL_REPLICATION_LONG
+bool replica_disk_migrator::migrate_replica_checkpoint(const replica_disk_migrate_request &req)
+{
+    error_code sync_checkpoint_err = _replica->get_app()->sync_checkpoint();
+    if (sync_checkpoint_err != ERR_OK) {
+        derror_replica("disk migration(origin={}, target={}) sync_checkpoint failed({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       sync_checkpoint_err.to_string());
+        reset_status();
+        return false;
+    }
+
+    error_code copy_checkpoint_err =
+        _replica->get_app()->copy_checkpoint_to_dir(_target_data_dir.c_str(), 0 /*last_decree*/);
+    if (copy_checkpoint_err != ERR_OK) {
+        derror_replica("disk migration(origin={}, target={}) copy checkpoint to dir({}) "
+                       "failed(error={}), the dir({}) will be deleted",
+                       req.origin_disk,
+                       req.target_disk,
+                       _target_data_dir,
+                       copy_checkpoint_err.to_string(),
+                       _target_replica_dir);
+        reset_status();
+        utils::filesystem::remove_path(_target_replica_dir);
+        return false;
+    }
+
+    return true;
+}
+
+// THREAD_POOL_REPLICATION_LONG
+bool replica_disk_migrator::migrate_replica_app_info(const replica_disk_migrate_request &req)
+{
+    replica_init_info init_info = _replica->get_app()->init_info();
+    error_code store_init_info_err = init_info.store(_target_replica_dir);
+    if (store_init_info_err != ERR_OK) {
+        derror_replica("disk migration(origin={}, target={}) stores app init info failed({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       store_init_info_err.to_string());
+        reset_status();
+        return false;
+    }
+
+    replica_app_info info(&_replica->_app_info);
+    std::string path = utils::filesystem::path_combine(_target_replica_dir, ".app-info");
```

Comment:
I think you can create a const variable for ".app-info", just like `kDataDirFolder`